runtime/interpreter/interpreter.go

```diff
@@ -2485,10 +2485,40 @@ func (interpreter *Interpreter) functionConditionsWrapper(
 
 					body = func() controlReturn {
 
+						/* pre and post condition wrappers "re-declare" the same
+						   parameters as are used in the actual body of the function.
+						   When these parameters are given resource-kinded arguments,
+						   this can trick the resource analysis into believing that these
+						   resources exist in two variables at once. This is not the case,
+						   however, as execution of the pre and post condition occurs strictly
+						   before and after execution of the body respectively.
+
+						   To prevent the analysis from reporting a false positive here, when
+						   we enter the body of the wrapped function, we invalidate any resources
+						   that were tranferred to parameters by the precondition block, and then
+						   restore them after execution of the body for use by the post-condition
+						   block.
+						*/
+
+						vars := make(map[ResourceKindedValue]*Variable)
+						for _, arg := range invocation.Arguments {
+							resourceKindedValue := interpreter.resourceForValidation(arg)
+							if resourceKindedValue != nil {
+								vars[resourceKindedValue] = interpreter.resourceVariables[resourceKindedValue]
+								interpreter.invalidateResource(resourceKindedValue)
+							}
+						}
 						// NOTE: It is important to actually return the value returned
 						//   from the inner function, otherwise it is lost
 
 						returnValue := inner.invoke(invocation)
+
+						/* Restore resources invalidated before execution of the body */
+						for val, variable := range vars { //nolint:maprangecheck
+							interpreter.invalidateResource(val)
+							interpreter.resourceVariables[val] = variable
+						}
+
```

Comment:
Great solution to the problem: Temporarily un-tracking the resources before the inner function starts tracking them, then restoring the tracking after the inner function finished makes sense 👍 

A few things:

- We usually use // comments everywhere 
- Maybe mention the code which starts tracking of the resource-kinded arguments, `bindParameterArguments` above
- Only allocate if necessary. Many functions have no resource-kinded arguments
- In the loop body, continue early (just like in functions, where we return early)
- Iteration using `range` over Go maps is non-deterministic, like CI warned about
  - Ignoring it is safe at the moment, but the body of loop may perform side-effects in the future (e.g. report errors), so this could become unsafe in the future
  - There is actually no need to use a map, as the map is only iterated over, it is not used for lookups
  - You can just e.g. use a slice of tuples, which is already ordered

```suggestion
						// Pre- and post-condition wrappers "re-declare" the same
						// parameters as are used in the actual body of the function,
						// see the use of bindParameterArguments at the start of this function wrapper.
						//
						// When these parameters are given resource-kinded arguments,
						// this can trick the resource analysis into believing that these
						// resources exist in multiple variables at once
						// (one for each condition wrapper + the function itself).
						//
						// This is not the case, however, as execution of the pre- and post-conditions
						// occurs strictly before and after execution of the body respectively.
						//
						// To prevent the analysis from reporting a false positive here,
						// when we enter the body of the wrapped function,
						// we invalidate any resources that were assigned to parameters by the precondition block,
						// and then restore them after execution of the wrapped function,
						// for use by the post-condition block.

						type argumentVariable struct {
							variable *Variable
							value    ResourceKindedValue
						}

						var argumentVariables []argumentVariable
						for _, argument := range invocation.Arguments {
							resourceKindedValue := interpreter.resourceForValidation(argument)
							if resourceKindedValue == nil {
								continue
							}

							argumentVariables = append(
								argumentVariables,
								argumentVariable{
									variable: interpreter.resourceVariables[resourceKindedValue],
									value:    resourceKindedValue,
								},
							)

							interpreter.invalidateResource(resourceKindedValue)
						}

						// NOTE: It is important to actually return the value returned
						//   from the inner function, otherwise it is lost

						returnValue := inner.invoke(invocation)

						// Restore the resources which were temporarily invalidated
						// before execution of the inner function

						for _, argumentVariable := range argumentVariables {
							value := argumentVariable.value
							interpreter.invalidateResource(value)
							interpreter.resourceVariables[value] = argumentVariable.variable
						}
```